src/redux/bookSlice.js

```diff
@@ -0,0 +1,28 @@
+/*eslint-disable*/
+import {createSlice} from '@reduxjs/toolkit';
+
+const initialBook = [];
+const bookSlice = createSlice({
+  name: 'book',
+  initialBook,
+  reducers: {
+    addBook: (state, action) => ({
+      ...state,
+      books: [
+        {
+          id: action.payload.id,
+          title: action.payload.title,
+          author: action.payload.author,
+          category: action.payload.category,
+        },
+      ],
```

Comment:
- [x] Kinldy, as the element in your state are objects, make sure the added book is also an object. as you may notice you are adding if it's an array or what ...
- [x] And also make sure the id for a specific book can be initialized automatically since there will be no input to add an `id` from the user.
```suggestion
        {
          id: Math.floor(Math.random() * 100),
          title: action.payload.title,
          author: action.payload.author,
          category: action.payload.category,
        },
```